Modified builders.py and Add frontend page integration

Salmathmullali opened the pull request:
Sir,

I’ve added the builder functionality and an additional frontend page (for expired zim function). The frontend page works as expected. However, I encountered an issue with the API, where it shows "Not Found" when trying to retrieve files. I tried to debug it but could not resolve the issue, and I am still new to working with APIs. I thought I should submit my changes and leave this part open for now.

I look forward to any feedback or suggestions on how to proceed with fixing the API issue.

Thanks.

kelson42 commented:
@Salmathmullali Thank you for your PR. Can you please tell which issue is fixed by your PR?

Salmathmullali commented:
Hi @kelson42,

thank you for your response! 


This PR is intended to address issue [#825] checking the ZIM file status before redirecting.

In my first PR, I received feedback from @audiodude suggesting I move the logic into builder.py (around line 227) and also include a frontend page that explains the expiry issue and provides a button to re-request the ZIM. I followed that advice and updated the code accordingly.

In this current PR:

I’ve modified builder.py as per the suggestion.
I’ve also added a new frontend page to show the “ZIM expired” message and provide a re-request button.

The frontend part is working fine when accessed directly. However, I ran into an issue where the API returns a “Not Found” error when trying to handle expired files. As a beginner, I wasn’t sure if the problem was with my routing or the API itself, so I wanted to submit my progress so far and ask for guidance on how to debug or structure this better.

I’d really appreciate any suggestions or pointers. I’m willing to keep improving this PR and learn as I go.

Thanks again!

audiodude commented on Pipfile:
If you need a newer version of `attrs`, please put the version number you need in this file. Also no need to change the ordering.

audiodude commented on wp1-frontend/src/components/ExpiredZim.vue:
```diff
@@ -0,0 +1,84 @@
+</script>
+
+<style scoped>
+.container {
```
We shouldn't need many, if any, component styles. Please use [Bootstrap](https://getbootstrap.com/docs/4.6/getting-started/introduction/) classes for styling.

audiodude commented on wp1-frontend/src/components/ExpiredZim.vue:
Please include a screenshot of this frontend page.

audiodude commented on wp1-frontend/src/main.js:
```diff
@@ -30,177 +31,61 @@ Vue.use(VueRouter);
-      title: () => BASE_TITLE + ' - ZIM file',
-    },
-  },
+  { path: '/', component: IndexPage, meta: { title: () => BASE_TITLE } },
```
Please restore the previous formatting of the routes array.

audiodude commented on wp1-frontend/src/main.js:
```diff
@@ -30,177 +31,61 @@ Vue.use(VueRouter);
 ];
 
 const router = new VueRouter({
+  mode: 'history',
```
What does this do and why is it necessary?

audiodude commented on wp1-frontend/src/main.js:
```diff
@@ -30,177 +31,61 @@ Vue.use(VueRouter);
-    } else {
-      return { x: 0, y: 0 };
-    }
+    if (savedPosition) return savedPosition;
```
Please restore previous formatting

audiodude commented on wp1-frontend/src/main.js:
```diff
@@ -30,177 +31,61 @@ Vue.use(VueRouter);
+
+
+router.beforeEach(async (to, from, next) => {
+  console.log(`Navigating to: ${to.path}`); 
```
nit: remove console.log

audiodude commented on wp1-frontend/src/main.js:
```diff
@@ -30,177 +31,61 @@ Vue.use(VueRouter);
 
-router.beforeEach((to, from, next) => {
+
+async function checkZimFileExpiration() {
```
Any function like this is not appropriate for `main.js`. The `main.js` file should just set up routes and initialize the Vue application. Logic like this should live in the component.

audiodude commented on wp1-frontend/src/main.js:
```diff
@@ -30,177 +31,61 @@ Vue.use(VueRouter);
+  console.log(`Navigating to: ${to.path}`); 
   document.title = to.meta.title(to);
+  
+  if (to.path !== '/expired-zim') {
```
Just as `checkZimFileExpiration` does not belong in this file, this block that calls it should also not be here.

audiodude commented on wp1-frontend/src/main.js:
```diff
@@ -30,177 +31,61 @@ Vue.use(VueRouter);
   next();
 });
 
 new Vue({
```
No need to modify this call.

audiodude commented on wp1/web/__init__.py:
```diff
@@ -1,24 +1,32 @@
 try:
-  #  The credentials module isn't checked in and may be missing
-  from wp1.credentials import ENV, CREDENTIALS
+    
```
Please restore 2-space indentation. You can run `pipenv run yapf -i wp1/web/__init__.py` to properly format the file. You should also consider installing the pre-commit hooks [mentioned in the README](https://github.com/openzim/wp1?tab=readme-ov-file#pre-commit-hooks), which will make sure that you cannot accidentally commit Python files with incorrect formatting.

audiodude commented on wp1/web/__init__.py:
```diff
@@ -1,24 +1,32 @@
-      return f(*args, **kwargs)
-    flask.abort(401, 'Unauthorized')
+  
+@app.route('/')
```
This looks like debug code and should be removed.

audiodude commented on wp1/web/__init__.py:
I'm not sure there are any significant changes to this file, besides changing the formatting and adding a debug function. Consider reverting it.

audiodude commented on wp1/web/builders.py:
It looks like all you did to this file was remove all the comments? Maybe you have a bad commit somewhere?

audiodude commented on wp1-frontend/src/components/ExpiredZim.vue:
```diff
@@ -0,0 +1,84 @@
+  methods: {
+    async checkZimStatus() {
+      try {
+        const response = await fetch('<WASABI_STORAGE_URL>', { method: 'HEAD' });
```
You still should not be doing this in the frontend, and the fact that you're using the literal string `<WASABI_STORAGE_URL>` demonstrates a lack of understanding of how the `fetch` method works.

Was this code copied from an LLM by any chance?

audiodude reviewed (changes requested):
This PR looks like it's not only incomplete, but doesn't even partially solve any stated problems. There are multiple significant issues that need to be addressed. Please test your code and make sure that it works fully before submitting another review request.

In particular, we will not be able to merge any code that says:

```
// TODO: Call API backend
```

Successful PRs must include complete solutions.

Finally, this PR doesn't include any tests. Full frontend and backend tests for all added/modified changes are necessary for any PRs.

Thanks!

Salmathmullali commented:
Sir,

Thank you so much for all the guidance and detailed feedback you've provided . I really appreciate the time and effort you put into helping me learn. 

To be transparent, I'm self-taught and currently learning on my own without a mentor. This is my first time contributing to a real-world project of this scale, and while I’ve been making mistakes, I’m committed to learning from them and improving. Your feedback is helping me understand not just the code, but also how to work within a team and follow best practices.

Regarding your question about LLM usage: yes, I’ve used it occasionally . Mainly for guidance on fixing specific errors or improving explanations . But the core understanding and effort are my own. My goal is to grow as a contributor and become a valuable part of this project. I may be new, but I’m putting in hard work to learn and contribute meaningfully.

I’ll now focus on addressing your comments in the PR and making the necessary corrections. Thank you again for the opportunity to learn from you.

Hopefully
Salmath Mullali

audiodude commented:
Hi Salmath,

First off, no need to address me as sir. :smile:

I appreciate your honesty, as well as your enthusiasm to learn and contribute. I hope you find my comments to be helpful, fair, and not (too) judgemental.

I'm looking forward to seeing your revisions and improvements to this PR and will gladly review it further once it's ready.

Salmathmullali commented:
Thanks for the feedback, really appreciate your guidance 🤝 I'll work on the changes and update soon!
